syntax="proto2";

package data;

message CardSet {
  enum CardSetId { universal = 0; general = 1; };
  optional CardSetId id = 1;
  }

message CardTag {
enum TagId {universal = 0; missile = 1; fighter = 2; decoy = 3;};
}

message CardCategory {
enum CategoryId {universal = 0; space = 1; crew = 2; tech = 3;};
}


message CardData {
enum CardId {
  universal = 0;
  basic_missile = 1;
  basic_janitor = 2;
  basic_laser_flurry = 3;
  basic_olga = 4;
  basic_emp = 5;
  basic_action_stations = 6;
  basic_mosquito = 7;
  basic_flare = 8;
  };

  enum SMSType { none = 0; science = 1; military = 2; support = 3;};
 optional   CardId id = 1;
 optional    string name = 2;
 optional  SMSType type = 3;
 optional uint32 system_id = 4;
 optional uint32 science_cost = 5;
 optional  uint32 military_cost = 6;
 optional uint32 support_cost = 7;
 repeated CardTag.TagId tags = 8;
 optional CardCategory.CategoryId category = 9;
 optional string description = 10;
 optional string fluff = 11;
 optional CardSet.CardSetId set = 12;
 optional int32 attack = 13;
 optional int32 health = 14;
 optional uint32 science_provide = 15;
 optional uint32 military_provide = 16;
 optional uint32 support_provide = 17;
}

message CardDataWrapper {
repeated CardData wrapped_msgs = 1; 
}